packages/stargate/src/signingstargateclient.ts

```diff
@@ -241,11 +252,43 @@ export class SigningStargateClient extends StargateClient {
   ): Promise<BroadcastTxResponse> {
     const withdrawMsg = {
       typeUrl: "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
-      value: MsgWithdrawDelegatorReward.fromPartial({ delegatorAddress: delegatorAddress, validatorAddress }),
+      value: MsgWithdrawDelegatorReward.fromPartial({
+        delegatorAddress: delegatorAddress,
+        validatorAddress: validatorAddress,
+      }),
     };
     return this.signAndBroadcast(delegatorAddress, [withdrawMsg], this.fees.withdraw, memo);
   }
 
+  public async sendIbcTokens(
+    senderAddress: string,
+    recipientAddress: string,
+    transferAmount: Coin,
+    sourcePort: string,
+    sourceChannel: string,
+    timeoutHeight?: Height,
+    /** timeout in seconds */
+    timeoutTimestamp?: number,
+    memo = "",
+  ): Promise<BroadcastTxResponse> {
+    const timeoutTimestampNanoseconds = timeoutTimestamp
+      ? Long.fromNumber(timeoutTimestamp * 1_000_000_000)
```

Comment:
This easily exceeds `MAX_SAFE_INTEGER`:

```
> const timestamp = (Date.now()/1000)
undefined
> (timestamp*1_000_000_000) / Number.MAX_SAFE_INTEGER
179.5539240475597
```

Let's convert `timeoutTimestamp` to string, append "000000000" and use Long.fromString

Reply:
What about `Long.fromNumber(timeoutTimestamp).multiply(1_000_000_000)`?

Reply:
@bartmacbartek This also applies to ts-relayer:

https://github.com/confio/ts-relayer/blob/main/src/lib/ibcclient.ts#L1348

Reply:
> What about `Long.fromNumber(timeoutTimestamp).multiply(1_000_000_000)`?

This is what you do when you know the Long API :)